src/app/pages/questions/containers/questions-page.component.ts

```diff
@@ -121,10 +134,42 @@ export class QuestionsPageComponent implements OnInit {
       .handleClinicalFollowUp(this.assessment, completedInClinic)
       .then(() => {
         this.updateDoneButton(false)
+        this.launchApp()
         return this.navCtrl.setRoot(HomePageComponent)
       })
   }
 
+  launchApp() {
+    if(!this.launchAppData) return
+    const options: AppLauncherOptions = {}
+
+    if(this.platform.is('ios')) {
+      options.uri = this.launchAppData.ios_uri
+    } else {
+      options.uri = this.launchAppData.android_uri
+    }
+    if ('?' in (options.uri as any)) {
+      options.uri += '&'
+    } else {
+      options.uri += '?'
+    }
+    options.uri += 'timestamp=' + this.task.timestamp
+
+    this.appLauncher.canLaunch(options)
+      .then((canLaunch: boolean) => {
+        if(canLaunch){
+          this.appLauncher.launch(options).then(()=>{
+            console.log('App launched')
+          }, (err)=>{
+            console.log('Error in launching app', err)
+          })
+        } else {
+          console.log('App cannot be launched')
+        }
+      })
+      .catch((error: any) => console.log('Error in checking if the app can launch', error))
```

Comment:
@blootsvoets I added some logs instead of showing the user that the uri is not reachable in the device. If the uri is not reachable or there is an error in opening the app, nothing happens.

Reply:
Would it be possible instead to already call the `canLaunch()` when initialising so the text can be changed to show the user does not support the package? Then the console can log the error if the `launch()` function fails.

Reply:
It's possible but the problem is how the user knows there is something remained after finishing. 
In this scenario, I think we should tell the user that if you press `finish button` an external app will open. 
1) Do you have any suggestion?
And there, if the app doesn't exist on the device (the uri is not reachable) there should be an alert.
2) I can set an alert as a toast or show alert from `alertService`. Which one do you think is better?

Reply:
Indeed, I think if the another app will be launched after finish, the button should show `Launch ...` or `Launch to finish` and the text should say this will launch ... app and finish the questionnaire.

As for alerts, I'd think a dedicated page would be more intuitive for people, but I'm not sure if there's a dedicated page after the finish page? If not, an alert on the home screen would be enough I suppose.